tests/gpu/test_gpu.py

```diff
@@ -0,0 +1,60 @@
+import os
+import pickle
+import unittest
+
+import GPUtil
```

Comment:
I'm glad we're not making this a general dependency for metal, since it (like our contrib stuff) is for a special use case and we don't want to have dependency bloat. Maybe as a help though, we could add a little tests/gpu/requirements.txt file (like we have in contrib/) so that people realize there are new requirements to run the code in this subdirectory.